File: backend/src/012-protocolos_web/SFTP/SSH_SCP/connect.rb
#!/usr/bin/ruby

require "net/ssh"  # gem install net-ssh
require "net/sftp" # gem install net-sftp

class SSH
    def initialize(username, password, address)
        @ssh = Net::SSH.start(address, username, :password => password)
    end

    def execute(command)
        output = @ssh.exec!(command)
    end
end

class SFTP
    def initialize(username, password, address)
        @sftp = Net::SFTP.start(address, username, :password => password)
    end

    def upload(filepath_in, filepath_out)
        @sftp.upload!(filepath_in, filepath_out)
    end

    def download(filepath_in, filepath_out)
        @sftp.download!(filepath_in, filepath_out)
    end
end

=begin
Fuentes:
https://github.com/net-ssh/net-ssh
https://github.com/net-ssh/net-sftp

=end

File: backend/src/012-protocolos_red/SFTP/SSH_SCP/connect.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# pip3 install paramiko
from paramiko import SSHClient, AutoAddPolicy, WarningPolicy

conn = SSHClient()
conn.set_missing_host_key_policy(AutoAddPolicy())

class SSH:
    """Conexión con una máquina remmota mediante SSH"""
    def __init__(self, username, password, address):
        conn.connect(address,username=username, 
                              password=password)
        self.ssh = conn

    def execute(self, command):
        stdin, stdout, stderr = self.ssh.exec_command(command)
        return (stdout.readlines(), stderr.readlines())

class SFTP:
    """Conexión con una máquina remota mediante SFTP"""
    def  __init__(self, username, password, address):
        conn.connect(address,username=username,
                              password=password)
        self.ftp = conn.open_sftp()

    def upload(self, filepath_in, filepath_out):
        self.ftp.put(filepath_in, filepath_out)
        return True

    def download(self, filepath_in, filepath_out):
        self.ftp.get(filepath_in, filepath_out)
        return True